### PR TITLE
doc/start: update linking conventions

### PR DESCRIPTION
**Pull request by zdover23**
Update linking conventions in doc/start/documenting-ceph.rst, as requested in
https://github.com/ceph/ceph/pull/49889#issuecomment-1657750905 by bluikko. This commit introduces a convention that could be read as inconsistent with other conventions mentioned on the page altered by this commit. Future PRs will attenuate this inconsistency.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/start/documenting-ceph.rst [hunk 1]
```diff
@@ -973,6 +973,16 @@ convention was preferred because it made the documents more readable in a
...
 the other. Use whichever convention makes the text easier to read.
 
+Using a part of a sentence as a hyperlink, `like this <docs.ceph.com>`_, is
+discouraged. The convention of writing "See X" is preferred. Here are some
+preferred formulations:
+
+#. For more information, see `docs.ceph.com <docs.ceph.com>`_.
+
+#. See `docs.ceph.com <docs.ceph.com>`_.
+
```
Comment (bluikko) [inline doc/start/documenting-ceph.rst:983]: Is it 3 empty lines here on purpose and not 2 like usually?

**Review by bluikko — APPROVED** [review]

File: doc/start/documenting-ceph.rst (hunk 1 above)
Comment (zdover23) [reply doc/start/documenting-ceph.rst:983]: It's not on purpose. Blank lines, being unrendered, are not of great concern.

Comment (anthonyeleven) [reply doc/start/documenting-ceph.rst:983]: My employer has GitLab commit hooks that beg to differ ;)

Comment (zdover23) [reply doc/start/documenting-ceph.rst:983]: Alright. I've removed the third blank line.

**Review by colemitchell — APPROVED** [review]

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/52912 - Reef backport
https://github.com/ceph/ceph/pull/52913 - Quincy backport

**Pull request merged**